src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/AttributeValueResolver.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\ValueResolver;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+/**
+ * Delegates value resolving of arguments bearing the {@see ValueResolver} attribute to the specified resolver.
+ *
+ * @author Mathieu Lechat <mathieu.lechat@les-tilleuls.com>
+ */
+final class AttributeValueResolver implements ValueResolverInterface
+{
+    public function __construct(private readonly ContainerInterface $resolvers)
+    {
+    }
+
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        $attributes = $argument->getAttributesOfType(ValueResolver::class);
+
+        if (!$attributes) {
+            return [];
+        }
+
+        $resolverName = $attributes[0]->name;
+        $resolver = $this->resolvers->get($resolverName);
+        if (!$resolver instanceof ValueResolverInterface) {
```

Comment:
This misses support for the deprecated `ArgumentValueResolverInterface` (which is not gone yet)